app/src/main/kotlin/GridWorld/GridworldMDP.kt

```diff
@@ -35,38 +41,39 @@ class GridworldMDP(
         return reward
     }
 
-    override fun transition(state: GridworldState, action: GridworldAction): IDistribution<GridworldState> {
+    override fun transition(state: GridworldState, action: GridworldAction): GridworldState {
         if (state.isTerminal) {
-            return UniformDistribution(listOf(state))
+            return state
         }
         else if (rewards.any { r -> r.equals(state)}) {
-            return UniformDistribution(listOf(GridworldState(state.x, state.y, true)))
+            return GridworldState(state.x, state.y, true)
         }
 
         // if target is out of bounds, return current state
         val targetNeighbour = state.ResolveNeighbour(action, xSize, ySize) ?:
-            return UniformDistribution(listOf(state))
+            return state
 
-        val allNeighbours = mutableListOf<ProbabilisticElement<GridworldState>>()
+        val allNeighbours = mutableListOf<GridworldState>()
```

Comment:
nit: call this nontargetNeighbours since it contains the possible results other than the target action.